### PR TITLE
Add utility class to simultaneously visualize multiple robot poses and DIRCON callback.

### PR DESCRIPTION
**Pull request by mposa**
Related to #103 

Visualizes multiple states simultaneously, also wraps into a DIRCON callback function.

Parses a URDF multiple times to achieve this, with the only option of welding one frame to the world. It would probably be better to simple work off of an existing MBP, if that were possible.

### CODE REVIEW

**Comment by RussTedrake** [comment]
Nice.  Related to https://github.com/RobotLocomotion/drake/issues/9728 .

File: systems/trajectory_optimization/hybrid_dircon.cc [hunk 1]
```diff
@@ -548,6 +548,28 @@ void HybridDircon<T>::ScaleKinConstraintSlackVariables(
...
+  auto my_callback = [this](const Eigen::Ref<const VectorXd>& vars) {
+    VectorXd vars_copy = vars;
+    Eigen::Map<MatrixXd> states(vars_copy.data(),
+        this->plant_.num_positions() + plant_.num_velocities(), this->N());
+
+    this->callback_visualizer_->DrawPoses(
+        states.block(0, 0, plant_.num_positions(), this->N()));
+  };
+
+  auto binding = AddVisualizationCallback(my_callback, x_vars());
```
Comment (yminchen) [inline systems/trajectory_optimization/hybrid_dircon.cc:570]: This binding doesn't seem to be used, so we can remove the assignment?

File: examples/Cassie/run_dircon_squatting.cc [hunk 2]
```diff
@@ -633,6 +633,8 @@ void DoMain(double duration, int max_iter, string data_directory,
     }
   }
 
+  trajopt->CreateVisualizationCallback("examples/Cassie/urdf/cassie_v2.urdf");
```
Comment (yminchen) [inline examples/Cassie/run_dircon_squatting.cc:636]: `cassie_v2.urdf` should be replaced by `cassie_fixed_springs.urdf`

Comment (yminchen) [reply examples/Cassie/run_dircon_squatting.cc:636]: Additionally, we can add `CreateVisualizationCallback` to the `run_dircon_walking` example (line 908) as well

File: systems/trajectory_optimization/hybrid_dircon.cc (hunk 1 above)
Comment (mposa) [reply systems/trajectory_optimization/hybrid_dircon.cc:570]: Done.

File: examples/Cassie/run_dircon_squatting.cc (hunk 2 above)
Comment (mposa) [reply examples/Cassie/run_dircon_squatting.cc:636]: Done.

**Comment by mposa** [comment]
I could also add an option to not show every knot point during TrajectoryOptimization (instead showing every T knot points). Would this be useful?

**Comment by mposa** [comment]
Will also add transparency as an option, once upstream Drake feature is implemented. 

https://github.com/RobotLocomotion/drake/issues/13126

**Comment by yminchen** [comment]
> I could also add an option to not show every knot point during TrajectoryOptimization (instead showing every T knot points). Would this be useful?

Yes this feature will be helpful, especially when we cannot change the transparency of the poses between start and end. 
I'm imagining after https://github.com/RobotLocomotion/drake/issues/13126 is fixed we can show non-transparent poses every T knot points, and show the rest of the poses with high transparency.

**Comment by yminchen** [comment]
> I'm imagining after [RobotLocomotion/drake#13126](https://github.com/RobotLocomotion/drake/issues/13126) is fixed we can show non-transparent poses every T knot points, and show the rest of the poses with high transparency.

Actually, maybe showing the rest of the poses with high transparency is not a good idea, since the poses are probably going to overlap a lot and we won't gain any information from most of the poses.

**Comment by mposa** [comment]
Added the option to display fractional poses, which I think is worthwhile. Note that it shows poses

(1, 1 + T, 1 + 2T, 1 + 3T, ..., N)

always showing the last pose. Take a look and see what you think.

**Comment by yminchen** [comment]
Another use case just occurred to me, I think it will be great to have an API which takes a list of the key frame indices? For example, if we have a multi-mode trajectory, and the key frames we want to see might not necessarily locate at `(1, 1 + T, 1 + 2T, 1 + 3T, ..., N)`

**Comment by yangwill** [comment]
LGTM! Having the option to specify the number of poses when calling `DrawPoses()` might be slightly more flexible but I don't see a dire need for that in the foreseeable future.

By the way, I have code in a branch to visualize the trajectory after saving the states at the knot points to a LCM trajectory, so you don't need to rerun the optimization problem just to visualize something. Could be useful later on.

**Pull request merged**